### PR TITLE
Refactored lesson 11.

### PR DESCRIPTION
**Pull request by 443n0511**
# [Issue No.11](https://github.com/kenmori/handsonFrontend/blob/master/work/markup/1.md#11)
```json
{ "data": [
  {
    "a": "bookmark",
    "img": "img/1.png",
    "alt": "画像１",
    "text": "ブックマーク"
  },
  {
    "a": "message",
    "img": "img/2.png",
    "alt": "画像２",
    "text": "メッセージ"
  }
]}
```
前回までのコードを生かして [fetch](https://developer.mozilla.org/ja/docs/Web/API/Fetch_API/Using_Fetch)を使ってデータを取得してください

取得したデータは前回と同じように表示してください
## [CodeSandBox](https://codesandbox.io/s/js-task11-selun?file=/app.js)<br>
Latest Commit:[c0d555f](https://github.com/443n0511/js-lesson/pull/24/commits/c0d555f296bfa792a28492b3fbb3b1b1649d3187)

### CODE REVIEW

File: lesson11/app.js [hunk 1]
```diff
@@ -1,48 +1,51 @@
...
-        return jsonData;
-    } catch (err) {
-        console.error(err);
-    }
+const createLoading = () => {
+  const loadingImage = document.createElement("img");
+  loadingImage.id = "lodingImage";
+  loadingImage.src = "./img/loading-circle.gif";
+  div.appendChild(loadingImage);
 };
```
Comment (443n0511) [inline lesson11/app.js:12]: Cut out the loading creation function.

File: lesson11/app.js [hunk 2]
```diff
@@ -1,48 +1,51 @@
...
 
-async function init() {
-    let data;
-    data = await getJsonData();
-    toDomCreateErements(data);
-}
-
+const removeLoading = () => document.getElementById("lodingImage").remove();
```
Comment (443n0511) [inline lesson11/app.js:14]: Cut out the loading deletion process.

File: lesson11/app.js [hunk 3]
```diff
@@ -1,48 +1,51 @@
...
+const getData = async () => {
+  try {
+    const response = await fetch(URL);
+    if (response.ok) return await response.json();
+    throw new Error(response.statusText);
+  } catch (e) {
+    console.log(e);
+    throw new Error(e);
+  }
+};
```
Comment (443n0511) [inline lesson11/app.js:25]: I created a function that just retrieves data.
If it fails, it throws an error as exception handling.

File: lesson11/app.js [hunk 4]
```diff
@@ -1,48 +1,51 @@
...
+  createLoading();
+  let listData;
+  try {
+    listData = await getData();
+  } catch (e) {
+    console.error(e.message);
+    div.textContent = `エラー内容:${e.message}`;
+  } finally {
+    removeLoading();
+  }
```
Comment (sae-github) [inline lesson11/app.js:37]: IMO
I think the loading will disappear when the error occurs and the process moves to the catch method, and the error message is displayed. removeLoading(), which is executed finally, will result in the element being null and an error will occur.

How about creating a new element and adding the error message?

File: lesson11/app.js [hunk 5]
```diff
@@ -1,48 +1,51 @@
...
-    try {
-        const response = await fetch(getJsonUrl);
-        const jsonData = await response.json();
-        return jsonData;
-    } catch (err) {
-        console.error(err);
-    }
+const createLoading = () => {
+  const loadingImage = document.createElement("img");
+  loadingImage.id = "lodingImage";
```
Comment (sae-github) [inline lesson11/app.js:9]: nits
```suggestion
  loadingImage.id = "loadingImage";
```

or

``` js
loadingImage.id = “js-loadingImage";
```

Comment (443n0511) [reply lesson11/app.js]: Whoops!
Thanks for pointing this out. I have corrected it.
[c0e272e](https://github.com/443n0511/js-lesson/pull/24/commits/c0e272e4b1181023c00df48b579ddf776b02264f)

File: lesson11/app.js (hunk 4 above)
Comment (443n0511) [reply lesson11/app.js:43]: Thank you very much. I have created a new function to create error messages.
In that function, I am trying to display the error message on the newly created element.
[57bd1c0](https://github.com/443n0511/js-lesson/pull/24/commits/57bd1c021db569b4f30b0312f8ced6db687c35b2)
```js
const createErrorMessage = (error) => {
  const errorMessage = document.createElement("p");
  errorMessage.textContent = `エラー内容：${error.message}`;
  div.appendChild(errorMessage);
  console.error(error.message);
};
```
[codesandbox](https://codesandbox.io/s/js-task11-selun?file=/app.js)

**Review by sae-github — APPROVED** [review]

File: lesson11/app.js [hunk 6]
```diff
@@ -1,48 +1,57 @@
...
-        const response = await fetch(getJsonUrl);
-        const jsonData = await response.json();
-        return jsonData;
-    } catch (err) {
-        console.error(err);
-    }
+const URL = "https://myjson.dit.upm.es/api/bins/3hf1";
+
+const template = (value) =>
+  `<li><a href="${value.to}"><img src="${value.img}" alt="${value.alt}">${value.text}</a></li>`;
```
Comment (hippopo-moo) [inline lesson11/app.js:5]: I thought the code was very readable, more declarative than procedural! 
I will presumptuously approve it!

**Pull request merged**